src/dynamic_programming/longest_common_subsequence.rs

```diff
@@ -0,0 +1,95 @@
+/// # Longest Common Subsequence
+
+/// `longest_common_subsequence(v1, v2)` Given two sequences find the longest common subsequence.
+///     If there are multiple common subsequences with the same maximum length, return any one of them.
+///
+/// Assumptions: 1 <= v1 <= 100
+///              1 <= v2 <= 100
+pub fn longest_common_subsequence(v1: Vec<&str>, v2: Vec<&str>) -> String {
+    let mut memo = vec![vec![0; v1.len() + 1]; v2.len() + 1];
+
+    for i in 1..=v2.len() {
+        for j in 1..=v1.len() {
+            let diagonal = memo[i - 1][j - 1];
+            let up = memo[i - 1][j];
+            let left = memo[i][j - 1];
+
+            if v2[i - 1] == v1[j - 1] {
+                memo[i][j] = 1 + diagonal;
+            } else {
+                memo[i][j] = usize::max(left, up);
+            }
+        }
+    }
+
+    let mut solution: Vec<&str> = vec![];
+    let mut i = v2.len();
+    let mut j = v1.len();
+
+    while i > 0 && j > 0 {
+        let _i = i - 1;
+        let _j = j - 1;
+
+        let curr_count = memo[i][j];
+        let curr_v2_letter = v2[_i];
+        let curr_v1_letter = v1[_j];
+
+        if curr_v1_letter == curr_v2_letter {
+            i = _i;
```

Comment:
This is nice :)

Just some more comments to tell what are we doing, to generate the longest sequence from the matrix will be a plus.